contracts/pangolin-periphery/PangolinBridgeMigrationRouter.sol

```diff
@@ -0,0 +1,358 @@
+        require(pairA != address(0), "PangolinBridgeMigrationRouter::liquidityPairFrom address 0");
+        require(pairA != address(0), "PangolinBridgeMigrationRouter::liquidityPairTo address 0");
+        require(pairA != pairB, "PangolinBridgeMigrationRouter::cant convert to the same liquidity pairs");
+        require(
+            IPangolinPair(pairA).token0() == IPangolinPair(pairB).token0() ||
+            IPangolinPair(pairA).token0() == IPangolinPair(pairB).token1() ||
+            IPangolinPair(pairA).token1() == IPangolinPair(pairB).token0() ||
+            IPangolinPair(pairA).token1() == IPangolinPair(pairB).token1(),
+            "PangolinBridgeMigrationRouter::Pair does not have one token matching"
+        );
+    }
+
+    /**
+     * @notice Internal function that implements the token migration
+     * @dev This function only checks if the expected balance was received, it doesn't check for migrator existance
+     * @param token The ERC20 token address that will be migrated
+     * @param amount The amount of the token to be migrated
+     */
+    function _migrateToken(
+        address token,
+        uint amount
+    ) internal {
+        IBridgeToken(bridgeMigrator[token]).swap(token, amount);
+        require(
+            IBridgeToken(bridgeMigrator[token]).balanceOf(address(this)) == amount,
+            "PangolinBridgeMigrationRouter::Didn't yield the correct amount"
+        );
+    }
+
+    /**
+     * @notice Front facing function that migrates the token
+     * @dev This function includes important checks
+     * @param token The ERC20 token address that will be migrated
+     * @param to The address of who's receiving the token
+     * @param amount The amount of the token to be migrated
+     * @param deadline The deadline limit that should be met, otherwise revert to prevent front-run
+     */
+    function migrateToken(
+        address token,
+        address to,
+        uint amount,
+        uint deadline
+    ) external ensure(deadline) {
+        require(
+            bridgeMigrator[token] != address(0),
+            "PangolinBridgeMigrationRouter::migrator not registered"
+        );
+        TransferHelper.safeTransferFrom(token, msg.sender, address(this), amount);
+        _migrateToken(token, amount);
+        TransferHelper.safeTransfer(bridgeMigrator[token], to, amount);
+    }
+
+    /**
+     * @notice Front facing function that migrates the liquidity, with permit
+     * @dev This function assumes someone else already gave approval
+     * @param liquidityPairFrom The pair address to be migrated from
+     * @param liquidityPairTo The pair address to be migrated to
+     * @param to The address that will receive the liquidity and the charge backs
+     * @param amount The amount of token liquidityPairFrom that will be migrated
+     * @param deadline The deadline limit that should be met, otherwise revert to prevent front-run
+     */
```

Comment:
missing documentation for params v, r and s

Reply:
I'm likely missing something, but can someone not bypass the permit by calling `migrateLiquidity` instead of `migrateLiquidityWithPermit?

Reply:
definitely, if they did that, it would require they called `approve` on the PGL contract approving the migrator. using permit allows some gas and another tx

Reply:
should update the docs to remove the description saying this assumes the contract was pre-approved

Reply:
Going to resolve this discussion as it's more of a doubt than a review itself, feel free to reopen if you want to further discuss @akly

Reply:
(forgot to mention, yes, the docstring is coming)